src/interpreter/object.cc

```diff
@@ -22,8 +22,16 @@ namespace verona::interpreter
   {
     rt::Descriptor::size = sizeof(VMObject);
     rt::Descriptor::trace = VMObject::trace_fn;
-    rt::Descriptor::trace_possibly_iso = VMObject::trace_fn;
-    rt::Descriptor::finaliser = VMObject::finaliser_fn;
+    rt::Descriptor::destructor = VMObject::destructor_fn;
```

Comment:
Could you use `is_trivially_destructible` trait here on VMObject to decide if we should have a destructor or not.  This will make the code more future proof.